### PR TITLE
Remove copy_imag and copy_real methods

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39099 Add view_as_real for complex tensors
* **#39065 Remove copy_imag and copy_real methods**

Differential Revision: [D21803939](https://our.internmc.facebook.com/intern/diff/D21803939)

### CODE REVIEW

File: aten/src/ATen/native/ReduceOps.cpp [hunk 1]
```diff
@@ -681,15 +681,15 @@ static Tensor &std_var_out(Tensor &result, const Tensor &self, IntArrayRef dim,
 
   if (at::isComplexType(self.scalar_type())){
     ScalarType dtype = c10::toValueType(get_dtype(result, self, {}, true));
-    Tensor real_in = self.copy_real().to(dtype);
+    Tensor real_in = at::real(self);
```
Comment (albanD) [inline aten/src/ATen/native/ReduceOps.cpp:684]: You lost the `.to(dtype)` here, is that expected? Why was it needed before?
Same for the others below.

**Review by albanD — COMMENTED** [review]
Code deletion 🎉

File: aten/src/ATen/native/ReduceOps.cpp (hunk 1 above)
Comment (anjali411) [reply aten/src/ATen/native/ReduceOps.cpp:684]: `copy_real` and `copy_imag` methods returned a complex tensor (`re+0j` / `im+0j`) before where the returned tensor's imaginary part was always zero and so `.to(equivalent_float_type)` was just added to return a float tensor ( as we are doing with `real` and `imag` now)

Comment (albanD) [reply aten/src/ATen/native/ReduceOps.cpp:684]: Sounds good!

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**